### PR TITLE
implementation for NSURL.standardizedURL and NSURL.filePathURL properties

### PR DESCRIPTION
**Pull request by mbvreddy**
Implementation for NSURL.standardizedURL and NSURL.filePathURL is added and enabled tests for NSURL.standardizedURL in TestNSURL.swift

### CODE REVIEW

File: Foundation/NSURL.swift [hunk 1]
```diff
@@ -28,6 +28,40 @@ private func _standardizedPath(_ path: String) -> String {
     return path
 }
 
+/* internal function to remove .. and . from path */
+internal func _standardizedURLPath(_ path: String) -> String {
```
Comment (parkera) [inline Foundation/NSURL.swift]: This should be private actually.

File: Foundation/NSURL.swift [hunk 2]
```diff
@@ -382,7 +416,9 @@ public class NSURL : NSObject, NSSecureCoding, NSCopying {
     
     /* A string constant for the "file" URL scheme. If you are using this to compare to a URL's scheme to see if it is a file URL, you should instead use the NSURL fileURL property -- the fileURL property is much faster. */
     public var standardizedURL: NSURL? {
-        NSUnimplemented()
+        let urlComponent = NSURLComponents(string: relativeString)
+        urlComponent!.path = _standardizedURLPath(urlComponent!.path!)
```
Comment (parkera) [inline Foundation/NSURL.swift]: Shouldn't we just return nil here if urlComponent is nil?

Comment (parkera) [inline Foundation/NSURL.swift]: Same if path is nil

**Comment by mbvreddy** [comment]
Right. I will change and squash it now.

**Comment by mbvreddy** [comment]
@parkera I've implemented the changes suggested.

**Comment by parkera** [comment]
@swift-ci please test

**Comment by mbvreddy** [comment]
I'm in process of resolving conflicts by re-basing and recompiling.

**Comment by mbvreddy** [comment]
resolved conflicts

**Comment by mbvreddy** [comment]
@parkera After rebasing, I changed the fix and considered your earlier suggestions. Please review.

**Comment by mbvreddy** [comment]
@parkera please review

**Comment by parkera** [comment]
@swift-ci please test

**Comment by parkera** [comment]
Thanks for the ping

**Pull request merged**